dtfs-central-api/src/services/portal/deal.service.ts

```diff
@@ -47,28 +47,35 @@ export class PortalDealService {
    * @param addGefDealCancelledActivityParams
    * @param addGefDealCancelledActivityParams.dealId - the deal
    * @param addGefDealCancelledActivityParams.author - the activity's author
+   * @param addGefDealCancelledActivityParams.cancellationIsInFuture - cancellation is in the future
    * @param addGefDealCancelledActivityParams.auditDetails - the users audit details
    */
   public static async addGefDealCancelledActivity({
     deal,
     author,
+    cancellationIsInFuture,
     auditDetails,
   }: {
     deal: TfmDeal;
     author: ActivityAuthor;
+    cancellationIsInFuture?: boolean;
```

Comment:
I think its a bit of a code smell to pass in a boolean flag, we could instead pass in `activityType`. 

It does make me wonder if this function is actually doing two things & should be split in two

Reply:
I don't agree, boolean is a native type, very common?

Using `activityType` instead won't work - we need to know if the cancellation is before or after "now" to set different activity values.

Reply:
My understanding is _Clean code_ encourages generally avoiding boolean arguments where possible because its a sign the function is doing more than one thing. It's not too big of an issue here but I think it'd be cleaner to either:
- create a second function `addGefCancellationPendingActivity`
- pass in a `label` parameter taking value `DEAL_CANCELLATION_PENDING` or `DEAL_CANCELLED` accordingly

Reply:
Thanks @MarRobSoftwire I don't entirely agree that a function shouldn't do more one thing, but do agree a `addGefCancellationPendingActivity` method would be cleaner, will update.

Reply:
@MarRobSoftwire I've improved this so we now have the following:

```ts
addGefDealCancellationPendingActivity
addGefDealCancelledActivity
```

And they both accept `dealId` and `dealType` instead of having to consume the whole deal object.